pkg/api/vlabs/validate.go

```diff
@@ -601,6 +601,16 @@ func (a *Properties) validateAddons() error {
 				if to.Bool(addon.Enabled) && a.HasCoreOS() {
 					return errors.New("flexvolume add-ons not currently supported on coreos distro. Please use Ubuntu")
 				}
+			case "kube-dns":
+				if to.Bool(addon.Enabled) {
+					maxVersion, err := semver.Make("1.12.0")
+					if err != nil {
+						return errors.New("could not validate version")
+					}
+					if sv.GT(maxVersion) {
```

Comment:
this should be is Greater or Equal (`.GE`), or is not Less Than (`! .LT`) since 1.12.0 is included in "not supported" cases.